packages/matter.js/src/device/AutoDevice.ts

```diff
@@ -0,0 +1,361 @@
+    ? S & StateType<C, R>
+    : T extends []
+    ? InstanceType<C>["state"]
+    : never;
+
+/**
+ * Methods related to state management.
+ */
+export interface StateManager<State extends BaseState> {
+    /**
+     * Access current attribute values.
+     */
+    get state(): State;
+
+    /**
+     * Update internal state.
+     * 
+     * @param state an object mapping attribute names to new values
+     */
+    set(changes: Partial<State>): void;
+
+    /**
+     * Process a validated write request from the fabric.  The default
+     * implementation applies all writes directly to internal state.
+     * 
+     * Derivatives should override this method to process state changes that
+     * have external dependencies.
+     * 
+     * @param state the attribute values to change
+     */
+    write(changes: Partial<State>): Promise<void>;
+}
+
+type Constructor = new (...args: any[]) => any;
+type UntypedClusterServer = <() => >;
+
+/**
+ * Construction options for AutoDevice.
+ */
+export type AutoDeviceOptions<State> = {
+    state?: Partial<State>;
+    endpointId?: number;
+}
+
+// Automatic wiring support class
+class WiringContext {
+    private serverHandlers: UntypedHandlers = {};
+
+    constructor(
+        // The cluster to wire
+        public cluster: ClusterModel,
+
+        // Hooks invoked at construction time to do perform setup
+        public initializers: Array<(this: AutoDevice, options: AutoDeviceOptions<BaseState>) => void>,
+
+        // Target prototype for device class
+        public deviceProto: any,
+
+        // Target prototype for state class
+        public stateProto: any,
+    ) {}
+
+    // Auto-wire state and device interfaces for a specific cluster
+    wire() {
+        this.cluster.attributes.forEach((attr) => {
+            this.wireAttributeState(attr);
+        });
+        this.wireAttributeEvents();
+
+        this.cluster.commands.forEach((command) => {
+            this.wireCommand(command);
+        });
+        
+        this.cluster.events.forEach((event) => {
+            this.wireEvent(event);
+        });
+
+        this.wireClusterServer();
+    }
+
+    // Implement the state interface
+    private wireAttributeState(attr: AttributeModel) {
+        const cluster = this.cluster;
+
+        // Getter
+        const descriptor = <PropertyDescriptor>{
+            get(this: BaseState) {
+                return this[DEVICE].attributeFor(cluster, attr).get();
+            }
+        };
+
+        // Setter
+        if (attr.fixed) {
+            descriptor.set = function(this: BaseState, value) {
+                this[DEVICE].attributeFor(cluster, attr).setLocal(value);
+                return true;
+            }
+        } else {
+            descriptor.writable = false;
+        }
+
+        // Add the property
+        Object.defineProperty(this.stateProto, attr.name, descriptor);
+
+        // If there's a default "get" handler, install it
```

Comment:
please let the attribute server being "the main place"  ... simply always use attribjute.getLocal ... I would not interfere that ddeep here because else it will be issues with changes later (and not only I might kill getters soon)

Reply:
OK cool, I wasn't sure what the get handlers were for anyway. I just added it because there were some defined in some of the cluster servers.

Reply:
These were for "attributes where values could be more easy generated dynamically then statically" ... but as written in discord this has big issues in subscriptions. I currently work on killing them. PR will come soon,